apps/framework-cli/src/mcp/compressed_map.rs

```diff
@@ -0,0 +1,659 @@
+            to: key.clone(),
+            connection_type: ConnectionType::Ingests,
+        });
+        compressed.add_connection(Connection {
+            from: key.clone(),
+            to: sync.target_table_id.clone(),
+            connection_type: ConnectionType::Ingests,
+        });
+    }
+}
+
+/// Add topic-to-topic sync processes to the compressed map
+fn add_topic_topic_syncs(compressed: &mut CompressedInfraMap, infra_map: &InfrastructureMap) {
+    for (key, sync) in &infra_map.topic_to_topic_sync_processes {
+        // Sync processes are derived from the source topic, use its source file
+        let source_file = infra_map
+            .topics
+            .get(&sync.source_topic_id)
+            .and_then(|t| t.metadata.as_ref())
+            .map(|m| extract_source_file(Some(m)))
+            .unwrap_or_default();
+
+        compressed.add_component(ComponentNode {
+            id: key.clone(),
+            component_type: ComponentType::TopicTopicSync,
+            name: format!("{} -> {}", sync.source_topic_id, sync.target_topic_id),
+            source_file,
+        });
+
+        // Add connections: source topic -> sync -> target topic
+        compressed.add_connection(Connection {
+            from: sync.source_topic_id.clone(),
+            to: key.clone(),
+            connection_type: ConnectionType::Transforms,
+        });
+        compressed.add_connection(Connection {
+            from: key.clone(),
+            to: sync.target_topic_id.clone(),
+            connection_type: ConnectionType::Produces,
+        });
+    }
+}
+
+/// Build a compressed infrastructure map from the full InfrastructureMap
+pub fn build_compressed_map(infra_map: &InfrastructureMap) -> CompressedInfraMap {
+    let mut compressed = CompressedInfraMap::new();
+
+    add_topics(&mut compressed, infra_map);
+    add_tables(&mut compressed, infra_map);
+    add_views(&mut compressed, infra_map);
+    add_api_endpoints(&mut compressed, infra_map);
+    add_functions(&mut compressed, infra_map);
+    add_sql_resources(&mut compressed, infra_map);
+    add_workflows(&mut compressed, infra_map);
+    add_web_apps(&mut compressed, infra_map);
+    add_topic_table_syncs(&mut compressed, infra_map);
+    add_topic_topic_syncs(&mut compressed, infra_map);
+
+    compressed
+}
+
+/// Build a resource URI for a component
+pub fn build_resource_uri(component_type: ComponentType, component_id: &str) -> String {
+    format!(
+        "moose://infra/{}/{}",
+        component_type.uri_segment(),
+        component_id
+    )
+}
+
+/// Parse a resource URI to extract component type and ID
+pub fn parse_resource_uri(uri: &str) -> Option<(ComponentType, String)> {
+    let prefix = "moose://infra/";
+    if !uri.starts_with(prefix) {
+        return None;
+    }
+
+    let path = &uri[prefix.len()..];
+    let parts: Vec<&str> = path.splitn(2, '/').collect();
+    if parts.len() != 2 {
+        return None;
+    }
```

Comment:
Nit: I would keep it as an iterator and just call next twice like `let component_type = split.next()?;`.